packages/react/src/pages/planning-calendar/planning-calendar.tsx

```diff
@@ -0,0 +1,182 @@
+import './planning-calendar.scss';
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { defaultCalendarListItems } from 'dx-template-gallery-data';
+
+import Button from 'devextreme-react/button';
+import Calendar from 'devextreme-react/calendar';
+import Scheduler, { Resource } from 'devextreme-react/scheduler';
+import SpeedDialAction from 'devextreme-react/speed-dial-action';
+import Tooltip from 'devextreme-react/tooltip';
+
+import { ViewType } from 'devextreme/ui/scheduler';
+import { useScreenSize } from '../../utils/media-query';
+
+import { useSchedulerLogic } from './use-scheduler-logic';
+
+import { CalendarList } from '../../components/calendar-list/calendar-list';
+import { LeftSidePanel } from '../../components/side-panel/left-side-panel';
+import { RightSidePanel } from '../../components/side-panel/right-side-panel';
+import { Agenda } from '../../components/agenda/agenda';
+import { TooltipContentTemplate } from '../../components/scheduler-tooltip/scheduler-tooltip';
+
+const views: ViewType[] = ['day', 'workWeek', 'month', 'agenda'];
+interface CalendarListItem {
+  id: number,
+  text: string,
+  color: string,
+  checkboxColor: string,
+}
+
+const onAppointmentFormOpening = (e) => {
+  const editor = e.form.getEditor('calendarId');
+  if (e.appointmentData.calendarId === undefined) {
+    editor.option('value', 0);
+  }
+};
+
+export const PlanningCalendar = () => {
+  const { isXSmall, isSmall } = useScreenSize();
+  const [calendarListItems] = useState(defaultCalendarListItems);
+  const {
+    agendaItems,
+    currentView,
+    date,
+    rightPanelOpen,
+    schedulerRef,
+    selectedAppointment,
+    tasks,
+    tooltipPosition,
+    tooltipRef,
+    createAppointment,
+    deleteCurrentAppointment,
+    editCurrentAppointment,
+    onCurrentViewChange,
+    onAppointmentClick,
+    onAppointmentTooltipShowing,
+    onCellModified,
+    onCellClick,
+    updateAgenda,
+    setDate,
+    setSelectedAppointment,
+    toggleRightPanelOpen,
+  } = useSchedulerLogic();
+
+  const onTodayClick = () => {
```

Comment:
Question: Why this logic differs from onSelectedDateChange?
If we could unify the logic, we could keep only one of these functions and don't export setDate from the hook